backend/src/cache/server/mutation.go

```diff
@@ -193,42 +192,14 @@ func getValueFromSerializedMap(serializedMap string, key string) string {
 	return value
 }
 
-func isValidPod(pod *corev1.Pod) bool {
-	annotations := pod.ObjectMeta.Annotations
-	if annotations == nil || len(annotations) == 0 {
-		log.Printf("The annotation of this pod %s is empty.", pod.ObjectMeta.Name)
-		return false
-	}
-	if !isKFPArgoPod(&annotations, pod.ObjectMeta.Name) {
-		log.Printf("This pod %s is not created by KFP.", pod.ObjectMeta.Name)
-		return false
-	}
+func isTFXPod(pod *corev1.Pod) bool {
 	containers := pod.Spec.Containers
-	if containers != nil && len(containers) != 0 && isTFXPod(&containers) {
-		log.Printf("This pod %s is created by TFX pipelines.", pod.ObjectMeta.Name)
-		return false
-	}
-	return true
-}
-
-func isKFPArgoPod(annotations *map[string]string, podName string) bool {
-	// is argo pod or not
-	if _, exists := (*annotations)[ArgoWorkflowNodeName]; !exists {
-		log.Printf("This pod %s is not created by Argo.", podName)
-		return false
+	if containers == nil || len(containers) == 0 {
+		log.Printf("This pod container does not exist.")
+		return true
```

Comment:
I may didn't well catch it. Why here it means it's TFX Pod?

Reply:
Logic is here: https://github.com/rui5i/pipelines/blob/8637972f41ffa8c69aa8db511fd60189b61f0346/backend/src/cache/server/mutation.go#L211. Basically we are checking if the main container's command contains "tfx/orchestration/kubeflow/container_entrypoint.py" which is a tfx specific entrypoint.